packages/ui/app/src/api-page/endpoints/EndpointContent.tsx

```diff
@@ -44,6 +54,26 @@ export const EndpointContent = React.memo<EndpointContent.Props>(function Endpoi
         [setHoveredResponsePropertyPath]
     );
 
+    const computeAnchor = useCallback(
+        (
+            attributeType: "request" | "response" | "path" | "query",
+            attribute:
+                | FernRegistryApiRead.ObjectProperty
+                | FernRegistryApiRead.PathParameter
+                | FernRegistryApiRead.QueryParameter
+        ) => {
+            let anchor = "";
+            if (isSubpackage(package_)) {
+                anchor += snakeCase(package_.urlSlug) + "_";
+            }
```

Comment:
not a big fan of type guards because they're very un-type-safe and can cause bugs down the line when the APIs change.

Also, this will only prepend the urlSlug of the parent subpackage. If the endpoint is nested several subpackages deep, do we need to prepend all the subpackages to the anchor?

Maybe we can just pass in `parentSlug` from the `ApiPackageContents` and use that:

```tsx
<Endpoint
  key={endpoint.id}
  endpoint={endpoint}
  parentSlug={slug}
  slug={joinUrlSlugs(slug, endpoint.urlSlug)}
/>
```

Reply:
Good catch. Ok, I'll see how we can do it with `parentSlug`. For the nested ones, yes we'll probably need to prepend since we need to ensure uniqueness for each anchor.